models/issues/issue_search.go

```diff
@@ -70,6 +72,15 @@ func (o *IssuesOptions) Copy(edit ...func(options *IssuesOptions)) *IssuesOption
 // applySorts sort an issues-related session based on the provided
 // sortType string
 func applySorts(sess *xorm.Session, sortType string, priorityRepoID int64) {
+	// Since this sortType is dynamically created, it has to be treated specially.
+	if strings.HasPrefix(sortType, ScopeSortPrefix) {
+		scope := strings.TrimPrefix(sortType, ScopeSortPrefix)
+		sess.Join("LEFT", "issue_label", "issue.id = issue_label.issue_id")
+		sess.Join("LEFT", "label", "label.id = issue_label.label_id and label.name LIKE ?", scope+"/%")
+		sess.Asc("label.exclusive_order").Desc("issue.id")
```

Comment:
(Hopefully the last concern in my mind)

Should we really use "LEFT JOIN" here?

1. Should the issues without the orders be sorted?
2. Different databases have different "null order" behavior:
    * https://stackoverflow.com/questions/7621205/sort-null-values-to-the-end-of-a-table
    * https://stackoverflow.com/questions/2051602/mysql-orderby-a-number-nulls-last

Reply:
It would be unusual and surprising for the sort to also act as a filter, so I think the answer to (1) is that they should be included, but below the better matching results.  That also serves as the answer to the LEFT JOIN, since it would otherwise act as a filter to only entries with the indicated labels.

The second question is a bit trickier, probably the most universal solution is to use COALESCE, something like:

```
-               sess.Asc("label.exclusive_order").Desc("issue.id")
+               // 9223372036854775807 == max bigint
+               sess.OrderBy("COALESCE(label.exclusive_order, 9223372036854775807) ASC").Desc("issue.id")
```